### PR TITLE
Allow specifying the serviceAccountName for the vitessbackupstorage subcontroller pod

### PR DESCRIPTION
**Pull request by bnu0**
Following up on a slack discussion yesterday with @enisoc, I think this is a fairly minimal change to get cross-namespace operator management working.

Currently, the following issue happens: 
- The operator attempts to fork its own pod into a new PodSpec, which is created [in the same namespace](https://github.com/planetscale/vitess-operator/blob/381260fcfddea3e6517b2ec1c616cca547b4d565/pkg/controller/vitessbackupstorage/reconcile_subcontroller.go#L82) as the VitessCluster CR.
- Indeed, we want it to be created in this namespace, because it might reference user-defined secrets needed by both the VitessCluster itself (such as the secret containing the `init_db.sql`) as well as secrets needed by backups (e.g. containing S3 credentials). We would want those to live alongside the cluster, not the operator.
- However, when forking the pod, it includes the original `serviceAccountName` as well as the auto-generated volumes and volumeMounts for mounting the token of that account. That account and/or token are unlikely to, and indeed should not, exist in some arbitrary other namespace where the VitessCluster lives.
- As a result, the pod cannot be created because it cannot find the serviceaccount or mount the token.

This change:
- Allows you to specify `VitessCluster.spec.backup.subcontroller.serviceAccountName`, which will be used to substitute a known serviceaccount in the target namespace.
- If unspecified (the default), then the current behavior is preserved.
- This gets reconciled into the `VitessBackupStorage.spec.subcontroller.serviceAccountName` field, which is the CR managed by the subcontroller.

What is missing:
- I was unable to run `gen-crd-api-reference-docs`; i am unsure what is different about my development environment, but i explicitly tried getting the explicit `@0.15.0` version, and still do not know why it is complaining.
- Documentation about the RBAC requirements of such a serviceaccount. Essentially the author of a VitessCluster CR in some arbitrary namespace  will need to create the serviceAccount, and make sure it has enough rights to do what the subcontroller requires. I've tried it with the same roles as the operator itself, and it works, however it only likely requires a subset of these capabilities and it would be good to figure out the minimum privileges needed and document those.

### CODE REVIEW

**Comment by bnu0** [comment]
I have tested this change successfully now; it seems like the minimum rbac `Role` yoou will need to bind to the `ServiceAccount` used for launching the vbs subcontroller pod is:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: ...
  namespace: ...
rules:
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
- apiGroups:
  - planetscale.com
  resources:
  - vitessbackups
  - vitessbackupstorages
  - vitessshards
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - planetscale.com
  resources:
  - vitessbackups
  - vitessbackupstorages/status
  verbs:
  - create
  - patch
  - update
  - delete
```

File: pkg/controller/vitessbackupstorage/reconcile_subcontroller.go [hunk 1]
```diff
@@ -130,6 +195,31 @@ func (r *ReconcileVitessBackupStorage) newSubcontrollerPodSpec(ctx context.Conte
 		return nil, fmt.Errorf("can't find operator container (name containing %q) in my own Pod", operatorContainerNameSubstring)
 	}
 
+	if vbs.Spec.PodServiceAccountName != "" {
```
[the remark this answers is not in the document]
Comment (bnu0) [reply pkg/controller/vitessbackupstorage/reconcile_subcontroller.go:198]: Yes the token suffix hash will be different even for a service account of the same name in a different namespace. Your suggestion makes sense. I will change it to always skip the auto-mounted tokens even if inheriting the serviceaccount from the operator pod.

**Comment by bnu0** [comment]
Thanks @enisoc, I've changed this per your suggestions.

**Review by enisoc — APPROVED** [review]
LGTM. Sorry for the delayed response. I already had a prototype of part of this before I found out you were working on it, so I wanted to rebase my changes on top of yours.

**Pull request merged**